app/models/manager_refresh/inventory_collection.rb

```diff
@@ -341,9 +342,10 @@ def initialize(model_class: nil, manager_ref: nil, association: nil, parent: nil
                    check_changed: nil, custom_manager_uuid: nil, custom_db_finder: nil, arel: nil, builder_params: {},
                    inventory_object_attributes: nil, unique_index_columns: nil, name: nil, saver_strategy: nil,
                    parent_inventory_collections: nil, manager_uuids: [], all_manager_uuids: nil, targeted_arel: nil,
-                   targeted: nil, manager_ref_allowed_nil: nil)
+                   targeted: nil, manager_ref_allowed_nil: nil, secondary_refs: {})
```

Comment:
Yeah I think putting these into one `:index` or `:indexes` makes sense, if we are going to pass in names of indexes I think `:additional_indexes` would be better than `:secondary_refs`

As for extracting `@data_index` I think that it would be better to keep it private